### PR TITLE
TST: tests for needs-test issues #12857 #12689

### LINKED ISSUE #6051: BUG: read multi-index column csv with index_col=False borks
http://stackoverflow.com/questions/21318865/read-multi-index-on-the-columns-from-csv-file

### LINKED ISSUE #9232: BUG: DataFrame constructor incorrect with Series input depending on name
DataFrame constructor example:

``` python
In [55]: x = pd.Series(range(5), name=1)

In [56]: y = pd.Series(range(5), name=0)

In [57]: pd.DataFrame(x, columns=[0])
Out[57]:
Empty DataFrame
Columns: [0]
Index: []

In [58]: pd.DataFrame(x, columns=[1])
Out[58]:
   1
0  0
1  1
2  2
3  3
4  4

```

and for y, the name and columns don't have to match for it to construct.

```
In [59]: pd.DataFrame(y, columns=[0])
Out[59]:
   0
0  0
1  1
2  2
3  3
4  4

In [60]: pd.DataFrame(y, columns=[1])
Out[60]:
   1
0  0
1  1
2  2
3  3
4  4
```

---

Using Pandas 0.15.2-1. After renaming a column in a dataframe, a .join operation fails.

``` python
import numpy as np,pandas as pd
df1=pd.DataFrame(range(1,1001)).rename(columns={0:1})
a=np.log(df1[1])
df2=df1.join(pd.DataFrame(a,columns=[0]))
```

The second column contains all NaN, instead of the values from _a_. The same can be accomplished by other means, but I believe the above should work. For example, the following works:

``` python
df1=pd.DataFrame(range(1,1001))
a=np.log(df1[0])
df2=df1.join(pd.DataFrame(a,columns=[2]))
```

### LINKED ISSUE #12687: COMPAT: breakage with just merged: https://github.com/numpy/numpy/pull/7325
see [here](https://travis-ci.org/pydata/pandas/jobs/117710492)
xref #12603 
#12644 

[numpy issue 7325](https://github.com/numpy/numpy/pull/7325)

cc @gfyoung 
cc @njsmith

### LINKED ISSUE #12689: ERR: invalid error reporting when comparing vs. None
The "in" function does not work as expected in regard to a data frame. Let's say we have the following data frame :

`df = pd.DataFrame(np.random.randn(6,4), columns=list('ABCD'))`

We can test this data frame :

```
df is None
False
```

However we cannot test its presence in a list : 

```
df in [None]
TypeError: Could not compare [None] with block values
```

We would expect to get just False in this case.

(Pandas version 0.17.1)

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #12689
- [x] closes #9232
- [x] closes #6051
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_repr_info.py [hunk 1]
```diff
@@ -27,6 +28,16 @@
...
 
 class TestDataFrameReprInfoEtc:
+    def test_repr_bytes_61_lines(self):
+        # GH#12857
+        lets = "ACDEFGHIJKLMNOP"
+        slen = 50
+        nseqs = 1000
+        words = [[random.choice(lets) for x in range(slen)] for _ in range(nseqs)]
+        df = pd.DataFrame(words).astype("S1")
+        assert (df.dtypes == "S1").all()
```
Comment (jreback) [inline pandas/tests/frame/test_repr_info.py]: hmm, do we actually allow this? meaning I *think* we *always* convert to object

File: pandas/tests/io/parser/test_header.py [hunk 2]
```diff
@@ -540,3 +540,32 @@ def test_multi_index_unnamed(all_parsers, index_col, columns):
...
+            (" Male", " L"),
+            (" Female", " R"),
+            (" Female", " R.1"),
+        ]
+    )
+    expected = DataFrame(
+        [[0.86, 0.67, 0.88, 0.78, 0.81], [0.86, 0.67, 0.88, 0.78, 0.82]], columns=mi
+    )
+
+    df1 = read_csv(StringIO(s1), header=[0, 1])
```
Comment (jreback) [inline pandas/tests/io/parser/test_header.py]: can you use self.read_csv (which tests all engines)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_repr_info.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_repr_info.py]: yah i thought this was a bit sketchy too.  ill pull this out so we can stick a pin in it while i mow down more of these "needs test" issues

File: pandas/tests/io/parser/test_header.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/io/parser/test_header.py]: done

**Review by mroeschke — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**